### ISSUE TITLE
fix: Datetime not selected after remove in Demo site

### ISSUE THREAD

**Issue by dungntw** [microsoft/fluentui-blazor#2197]
# 🐛 Bug Report

Datetime not selected after remove

https://github.com/microsoft/fluentui-blazor/assets/119403879/54874cdf-8115-4a6e-a6c9-1957ec59ac5a

I also checked in
- Form -> Overview
- Components -> DateTime -> Default Date Picker

## 🌍 Your Environment

* OS & Device: Windows on PC
* Browser Microsoft Edge, Google Chrome, Mozilla FireFox
* .NET and Fluent UI Blazor library Version 4.8.0

**Comment by SquaredLoops** [comment]
This **USED** to work and I think the bug was introduced as part of #2176 (certainly something in 4.8.0 release).

I know this because I have just spent the last couple of hours trying to figure out why my date picker that used to work has now stopped. My date picker is blank / empty on page load and a mandatory field, so was very obvious when it broke as the flow could no longer work.

When I looked in GitHub I saw this change, and this bug. When I looked at my VS project (that was created from the Fluent UI template) I noticed it references `4.*-*` so picked up this change. To fix it locally I forced to `4.7.2` so change your `csproj` files...

From:
```
<PackageReference Include="Microsoft.FluentUI.AspNetCore.Components" Version="4.*-* " />
```

To:
```
<PackageReference Include="Microsoft.FluentUI.AspNetCore.Components" Version="4.7.2 " />
```

I hope this can help others whose controls have suddenly started behaving strangely. I should have checked from the outset that my versions were pinned, so my fault, but a timely reminder for everyone to check! 😄
